### PR TITLE
[TS SDK] add User-Agent header

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Add User-Agent header to rest api calls 
### Test Plan
test are passing
and
```
2023-03-30T23:23:08.657964Z [api-18] DEBUG api/src/log.rs:53 {"elapsed":"835.083µs","method":"GET","path":"/v1/accounts/0x1","remote_addr":"127.0.0.1:65273","status":200,"user_agent":"aptos-ts-sdk / 1.7.2"}
```

### CODE REVIEW

File: ecosystem/typescript/sdk/src/version.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// hardcoded for now, we would want to have it injected dynamically
+export const VERSION = "1.7.2";
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/src/version.ts:2]: Please add to release instructions

**Review by gregnazario — APPROVED** [review]

File: ecosystem/typescript/sdk/src/providers/aptos_client.ts [hunk 2]
```diff
@@ -85,6 +86,7 @@ export class AptosClient {
     } else {
       conf.WITH_CREDENTIALS = true;
     }
+    conf.HEADERS = { ...conf.HEADERS, "User-Agent": `aptos-ts-sdk / ${VERSION}` };
```
Comment (banool) [inline ecosystem/typescript/sdk/src/providers/aptos_client.ts:89]:
```suggestion
    conf.HEADERS = { ...conf.HEADERS, "User-Agent": `aptos-ts-sdk/${VERSION}` };
```

File: ecosystem/typescript/sdk/src/version.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// hardcoded for now, we would want to have it injected dynamically
```
Comment (banool) [inline ecosystem/typescript/sdk/src/version.ts:1]: Can't you do something like this?
```
import {version} from './package.json';
```

**Review by banool — CHANGES_REQUESTED** [review]

File: ecosystem/typescript/sdk/src/version.ts (hunk 3 above)
Comment (0xmaayan) [reply ecosystem/typescript/sdk/src/version.ts:1]: not that easy, package.json is a configuration file and It doesn't contain any executable code that can be imported or used in the application, we need to do something like [this](https://github.com/alchemyplatform/alchemy-sdk-js/blob/main/inject-version.js) or [this](https://github.com/MystenLabs/sui/blob/main/sdk/typescript/genversion.mjs).
Also `import` used in esm modules while sdk supports both cjs and esm

Comment (banool) [reply ecosystem/typescript/sdk/src/version.ts:1]: Okay unfortunate, let's roll with this for now.

**Review by banool — APPROVED** [review]

**Pull request merged**